app/controllers/name_controller/create_and_edit_name.rb

```diff
@@ -173,8 +183,10 @@ def perform_change_existing_name
       any_changes = true
     end
     # Update ancestors regardless whether name changed; maybe this will add
-    # missing ancestors in case database is messed up
-    update_ancestors
+    # missing ancestors in case database is messed up. But don't update
+    # ancestors if non-admin is changing locked namge because that would create
+    # bogus name and ancestors if @parse.search_name differs from @name
+    update_ancestors if name_unlocked?
```

Comment:
1. The "bogus name" creation in the code comments cannot happen through the UI, but could happen via a crafted url or -- I suppose -- the API.
2. **Question**: Do we still need to update ancestors regardless whther the Name changed?

Reply:
This feels safe and correct to me.  DB can always get messed up.  API should be written in such a way as to prevent that, and the UI too.  But corruption happens.  (Lesser known version of the second law of thermodynamics! :)